GAE/src/com/gallatinsystems/survey/domain/WebForm.java

```diff
@@ -25,17 +25,26 @@
 
     public static Set<String> unsupportedQuestionTypes(){
         Set<String> unsupportedTypes = new HashSet<String>();
-        unsupportedTypes.add(Question.Type.CASCADE.toString());
         unsupportedTypes.add(Question.Type.GEOSHAPE.toString());
         unsupportedTypes.add(Question.Type.SIGNATURE.toString());
         unsupportedTypes.add(Question.Type.CADDISFLY.toString());
         return unsupportedTypes;
     }
 
-    public static boolean validWebForm(final List<Question> questions){
-        
-        List<Question> validQuestions = questions.stream().filter(i -> !unsupportedQuestionTypes().contains(i.getType().toString())).collect(Collectors.toList());
+    public static boolean validQuestionGroups(final Survey survey){
+        return survey.getQuestionGroupMap().values().stream().filter(i -> i.getRepeatable()).collect(Collectors.toList()).size() == 0;
+    }
 
+    public static Boolean validSurveyGroup(final SurveyGroup surveyGroup){
```

Comment:
So this one is a bit more subtle. It should be that webforms doesn't support monitoring forms. I.e. in a survey you have registration form and monitoring forms.  We should allow a registration form to be filled in.  Registration form id is === `SurveyGroup.newLocaleSurveyId` @marvinkome I guess this is already done on UI side?

Reply:
Yeah on the UI, monitoring surveys aren't supported

Reply:
Monitoring surveys or monitoring _forms_?

Reply:
👀 monitoring survey groups

Reply:
Should be the same... in a monitoring Survey(Group) you have two kinds of forms, registration and monitoring forms. Registration forms can be shared via webforms, monitoring forms cannot be shared. Registration form id is found by looking at the `SurveyGroup.newLocaleSurveyId` property